src/core/datastores/persisters/key-value-persister.js

```diff
@@ -0,0 +1,67 @@
+const _cache = {};
+
+export class KeyValuePersister {
+  /** @type {Boolean} */
+  _cacheEnabled;
+  /** @type {Number} */
+  _ttl;
+
+  // TODO: implement TTL, make _cache a constructor argument?
+  constructor(cacheEnabled = true, ttl = Infinity) {
+    this._cacheEnabled = cacheEnabled;
+    this._ttl = ttl;
+  }
+
+  readEntities(key) {
```

Comment:
We need to add some validation that will throw an `Error` if the `key` arguments is `null` or `undefined`.